fabric-client-gametest-api-v1/src/client/java/net/fabricmc/fabric/api/client/gametest/v1/ClientGameTestContext.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.commons.lang3.function.FailableFunction;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.SharedConstants;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.Screen;
+
+/**
+ * Context for a client gametest containing various helpful functions and functions to access the game. Functions in
+ * this class can only be called on the client gametest thread.
+ */
+@ApiStatus.NonExtendable
+public interface ClientGameTestContext {
+	/**
+	 * Used to specify that a wait task should have no timeout.
+	 */
+	int NO_TIMEOUT = -1;
+
+	/**
+	 * The default timeout for wait tasks.
+	 */
+	int DEFAULT_TIMEOUT = 10 * SharedConstants.TICKS_PER_SECOND;
+
+	/**
+	 * Runs a single tick and waits for it to complete.
+	 */
+	void waitTick();
+
+	/**
+	 * Runs {@code ticks} ticks and waits for them to complete.
+	 *
+	 * @param ticks The amount of ticks to run
+	 */
+	void waitTicks(int ticks);
+
+	/**
+	 * Waits for a predicate to be true. Fails if the predicate is not satisfied after {@link #DEFAULT_TIMEOUT} ticks.
+	 *
+	 * @param predicate The predicate to check
+	 */
+	void waitFor(Predicate<MinecraftClient> predicate);
+
+	/**
+	 * Waits for a predicate to be true. Fails if the predicate is not satisfied after {@code timeout} ticks. If
+	 * {@code timeout} is {@link #NO_TIMEOUT}, there is no timeout.
+	 *
+	 * @param predicate The predicate to check
+	 * @param timeout The number of ticks before timing out
+	 */
+	void waitFor(Predicate<MinecraftClient> predicate, int timeout);
+
+	/**
+	 * Waits for the given screen class to be shown. If {@code screenClass} is {@code null}, waits for the current
+	 * screen to be {@code null}. Fails if the screen does not open after {@link #DEFAULT_TIMEOUT} ticks.
+	 *
+	 * @param screenClass The screen class to wait to open
+	 */
+	void waitForScreen(@Nullable Class<? extends Screen> screenClass);
+
+	/**
+	 * Opens a {@link Screen} on the client.
+	 *
+	 * @param screen The screen to open
+	 * @see MinecraftClient#setScreen(Screen)
+	 */
+	void setScreen(@Nullable Screen screen);
+
+	/**
+	 * Presses the button in the current screen whose label is the given translation key. Fails if the button couldn't
+	 * be found.
+	 *
+	 * @param translationKey The translation key of the label of the button to press
+	 */
+	void clickScreenButton(String translationKey);
+
+	/**
+	 * Presses the button in the current screen whose label is the given translation key, if the button exists. Returns
+	 * whether the button was found.
+	 *
+	 * @param translationKey The translation key of the label of the button to press
+	 * @return Whether the button was found
+	 */
+	boolean tryClickScreenButton(String translationKey);
+
+	/**
+	 * Takes a screenshot after waiting 1 tick (for a frame to render) and saves it in the screenshots directory.
+	 *
+	 * @param name The name of the screenshot
+	 */
+	void takeScreenshot(String name);
```

Comment:
Would it be handy to return a `Path` to the screenshot if someone does want to do something with it during the test? Or maybe leave this for later PR where we provide some utils for inspecting the screenshot.